### ISSUE TITLE
[TypeScript fetch] doubling options

### ISSUE THREAD

**Issue by ElfenLiedGH** [swagger-api/swagger-codegen#8067]
##### Swagger-codegen version

2.3.1
##### Swagger declaration file content or url

```yaml
  /Companies/change-stream:
    post:
      tags:
        - Company
      summary: Create a change stream.
      operationId: Company.createChangeStream__post_Companies_change-stream
      parameters:
        - name: options
          in: formData
          required: false
          type: string
          format: JSON
      responses:
        '200':
          description: Request was successful
          schema:
            type: file
      deprecated: false
    get:
      tags:
        - Company
      summary: Create a change stream.
      operationId: Company.createChangeStream__get_Companies_change-stream
      parameters:
        - name: options
          in: query
          required: false
          type: string
          format: JSON
      responses:
        '200':
          description: Request was successful
          schema:
            type: file
      deprecated: false
```
##### Steps to reproduce

https://editor.swagger.io/#

menu generate client->typescript fetch

##### Suggest a fix/enhancement

generated code

        /**
         * 
         * @summary Create a change stream.
         * @param {string} [options] 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        companyCreateChangeStreamPostCompaniesChangeStream(options?: string, options?: any): (fetch?: FetchAPI, basePath?: string) => Promise<any> {
            const localVarFetchArgs = CompanyApiFetchParamCreator(configuration).companyCreateChangeStreamPostCompaniesChangeStream(options, options);
            return (fetch: FetchAPI = portableFetch, basePath: string = BASE_PATH) => {
                return fetch(basePath + localVarFetchArgs.url, localVarFetchArgs.options).then((response) => {
                    if (response.status >= 200 && response.status < 300) {
                        return response.json();
                    } else {
                        throw response;
                    }
                });
            };
        },

swagger api's generated auto by loopback v3

**Comment by macjohnny** [comment]
@ElfenLiedGH this is because by default you can pass a parameter `options` with a configuration object. A quick solution to your problem is to rename your `options` to e.g. `streamOptions`.
this issue is somehow similar to https://github.com/swagger-api/swagger-codegen/issues/7911, since we have naming conflicts with kind-of reserved keywords / names.

**Comment by ElfenLiedGH** [comment]
@macjohnny it's autogenerated api from IBM strongloop, i can't change any format

**Comment by macjohnny** [comment]
@ElfenLiedGH would you like to give it a try and implement a fix similar to #7911 ?
